src/resource.js

```diff
@@ -71,12 +72,19 @@ export default class Resource {
    * @returns {String|Array|Object}
    */
   get source() {
```

Comment:
As for relative paths - it should be addresed on the spec level first. It already has a restriction of usage of dot-paths, but it is not enough (forbids usage only in the start of that path)